src/pydvl/influence/array.py

```diff
@@ -0,0 +1,390 @@
+
+        Implement this method to define how a sequence of tensors, provided by a generator,
+        should be combined. Suitable for straightforward scenarios where
+        tensors are yielded one by one.
+        """
+
+
+class ListAggregator(TensorAggregator):
+    """
+    An aggregator that collects tensors into list structures.
+    """
+
+    def aggregate_from_nested_generators(
+        self,
+        nested_generators_of_tensors: Generator[
+            Generator[TensorType, None, None], None, None
+        ],
+    ):
+        """
+         Aggregates tensors from a nested generator structure into a list of lists. Each inner generator
+         is converted into a list of tensors, resulting in a nested list structure.
+
+         Args:
+             nested_generators_of_tensors: A generator of generators, where each inner generator yields
+                 TensorType objects.
+
+        Returns:
+            A list of lists, where each inner list contains tensors returned from one of the inner generators.
+        """
+        return [list(tensor_gen) for tensor_gen in nested_generators_of_tensors]
+
+    def aggregate_from_generator(
+        self, tensor_generator: Generator[TensorType, None, None]
+    ):
+        """
+        Aggregates tensors from a single-level generator into a list. This method simply collects
+        each tensor emitted by the generator into a single list.
+
+        Args:
+            tensor_generator: A generator that yields TensorType objects.
+
+        Returns:
+            A list containing all the tensors provided by the tensor_generator.
+        """
+        return [t for t in tensor_generator]
+
+
+class LazyArrayProvider(ABC):
+    """
+    Abstract base class representing a provider of lazily evaluated, array-like data. This class
+    is tailored for situations where data is not immediately computed or loaded, but is instead
+    generated or processed on-demand. The 'array-like' aspect refers to the data structure or
+    format rather than the ability to perform typical array operations like indexing.
+    """
+
+    @abstractmethod
+    def compute(self, tensor_aggregator: Optional[TensorAggregator] = None):
+        """
+        Triggers the computation or retrieval of the data, optionally aggregating the results using
+        a specified TensorAggregator. This method is responsible for initiating the actual data
+        processing or fetching tasks.
+
+        Args:
+            tensor_aggregator: An optional aggregator that specifies
+                how the results should be combined or processed.
+
+        Returns:
+            The result of the computation or data retrieval, processed as specified by the
+            tensor_aggregator, if provided. The exact return type and format depend on the
+            implementation and the aggregator used.
+        """
+
+    @abstractmethod
+    def to_zarr(
+        self,
+        path_or_url: str,
+        numpy_converter: NumpyConverter,
```

Comment:
And again: the only type supported is `NumpyConverter` => don't need it in the name.